pkg/ebpf/c/tracee.bpf.c

```diff
@@ -3172,34 +3147,61 @@ static __always_inline void invoke_print_syscall_table_event(event_data_t *data)
         syscall_num = (u64) *syscall_num_p;
         syscall_addr = READ_KERN(syscall_table_addr[syscall_num]);
         if (syscall_addr == 0) {
-            return;
+            return 0;
         }
         syscall_address[i] = syscall_addr;
     }
-    save_u64_arr_to_buf(data, (const u64 *) syscall_address, monitored_syscalls_amount, 0);
-    events_perf_submit(data, PRINT_SYSCALL_TABLE, 0);
+    save_u64_arr_to_buf(&data, (const u64 *) syscall_address, monitored_syscalls_amount, 0);
+    return events_perf_submit(&data, PRINT_SYSCALL_TABLE, 0);
 }
 
-SEC("kprobe/security_file_ioctl")
-int BPF_KPROBE(trace_tracee_trigger_event)
+static __always_inline void invoke_fetch_network_seq_operations_event(struct pt_regs *ctx,
+                                                                      unsigned long struct_address)
 {
     event_data_t data = {};
-
     if (!init_event_data(&data, ctx))
-        return 0;
+        return;
+
+    struct seq_operations *seq_ops = (struct seq_operations *) struct_address;
+    u64 show_addr = (u64) READ_KERN(seq_ops->show);
+    if (show_addr == 0) {
+        return;
+    }
+
+    u64 start_addr = (u64) READ_KERN(seq_ops->start);
+    if (start_addr == 0) {
+        return;
+    }
 
-    unsigned int cmd = PT_REGS_PARM2(ctx);
-    if ((cmd & IOCTL_FETCH_SYSCALLS) == IOCTL_FETCH_SYSCALLS &&
-        data.config->tracee_pid == data.context.task.host_pid) {
-        invoke_print_syscall_table_event(&data);
+    u64 next_addr = (u64) READ_KERN(seq_ops->next);
+    if (next_addr == 0) {
+        return;
     }
 
-    if ((cmd & IOCTL_HOOKED_SEQ_OPS) == IOCTL_HOOKED_SEQ_OPS &&
-        data.config->tracee_pid == data.context.task.host_pid) {
-        unsigned long struct_address = PT_REGS_PARM3(ctx);
-        invoke_fetch_network_seq_operations_event(&data, struct_address);
+    u64 stop_addr = (u64) READ_KERN(seq_ops->stop);
+    if (stop_addr == 0) {
+        return;
     }
+    u64 seq_ops_addresses[NET_SEQ_OPS_SIZE + 1] = {
+        (u64) seq_ops, show_addr, start_addr, next_addr, stop_addr};
+    save_u64_arr_to_buf(&data, (const u64 *) seq_ops_addresses, 5, 0);
+    events_perf_submit(&data, PRINT_NET_SEQ_OPS, 0);
+}
+
+SEC("uprobe/trigger_seq_ops_event")
+int uprobe_seq_ops_trigger(struct pt_regs *ctx)
+{
+#if defined(__TARGET_ARCH_x86)
+    uint64_t *address_array = ((void*)ctx->sp);
+#else
+    uint64_t *address_array = ((void*)ctx->sp)+16;
```

Comment:
Where does this 16 comes from? Is it something specific to ARM64?
If so, better if we add if defined(bpf_target_arm64)

Reply:
Yep, is this a calling convention for the uprobe eBPF programs contexts in both arches ? Did you find it by increasing the offset or is it documented somewhere ? A comment here would be nice as well.

Reply:
So it is documented but I had to do some testing to figure which argument goes where.
This is the documentation:
https://go.googlesource.com/go/+/refs/heads/dev.regabi/src/cmd/compile/internal-abi.md

I'll add a comment

Reply:
Thanks for the pointers for the golang calling convention (as a comment).